fea-rs/src/parse/grammar/metrics.rs

```diff
@@ -131,6 +146,138 @@ fn eat_metric(parser: &mut Parser, recovery: TokenSet) -> bool {
     true
 }
 
+// $pad or ${pad * 2 + 5}
+pub(crate) fn expect_glyphs_number_value(parser: &mut Parser, recovery: TokenSet) -> bool {
+    parser.in_node(AstKind::GlyphsNumberValueNode, |parser| {
+        assert!(parser.eat(Kind::Dollar));
+        if parser.eat_remap(Kind::Ident, AstKind::GlyphsNumberIdent) {
+            return true;
+        }
+        if parser.matches(0, Kind::LBrace) {
+            let recovery = recovery.add(Kind::RBrace).add(Kind::Dollar);
+            parser.in_node(AstKind::GlyphsNumberValueExprNode, |parser| {
+                parser.eat_raw();
+                let looks_okay = eat_expr_body(parser);
+                if !looks_okay {
+                    parser.eat_until(recovery);
+                }
+                parser.expect_recover(Kind::RBrace, recovery) && looks_okay
+            })
+        } else {
+            parser.err_recover("expected '$ident' or '${predicate}'", recovery);
+            false
+        }
+    })
+}
+
+// generally just eat values and operators; we will validate later
+// the one special case is '2-2' which we need to ensure gets parsed as (NUM)(-)(NUM)
```

Comment:
Should we have a test case for something like `2 - -2` ?

Reply:
Can we negate named values, e.g. `-padding`

Reply:
> Should we have a test case for something like `2 - -2` ?

There's a unit test for this in `glyphsapp_syntax_ext.rs`.

> Can we negate named values, e.g. `-padding`

no, we don't parse this.